### ISSUE TITLE
Production infra plan

### ISSUE THREAD

**Issue by ryanblock** [covidatlas/coronadatascraper#236]
## Brief

As we prepare to build what's currently called `CDS` into a more expansive end-user facing web application and set of APIs, we will begin moving this project into cloud infra (AWS).

Infrastructure-as-code will be provisioned via Architect, and crawling activity will become increasingly parallelized, writing back to the de facto blob store (S3).

## Objectives

- Complete a move onto AWS **without**:
  - A rewrite
  - Any interruption in current service
  - Adding any significant additional ceremony or barriers to entry for existing and new developers

## Plan phases

1. Move crawler logic into new crawler event directory (`src/events/crawler`)
   - No changes to local workflows will be necessitated
   - e.g. keep using `yarn start --location "Ventura County, CA, USA"`
   - https://github.com/lazd/coronadatascraper/pull/224
2. Begin building out AWS-specific logic
   - Write cache data to S3
   - This in addition to the current workflows continuing to operate and publish to GitHub
   - (discovery)
3. Activate scheduled crawling events
   - Start writing data to S3 / database in parallel with writing to git via GitHub Actions workflows
4. Build out new log egress
   - Details tbd / discover - fire to Slack?
5. Update and improve local workflows
   - Reduce surface area and commands required to build additional scrapers
     - Ideally `yarn dev` is the only command ever needed by a crawler or scraper dev to do their work
     - Begin phasing out git submodules by automatically pulling down cache data from S3 into local dev environment
6. Update GitHub Actions 
   - Update `deploy` actions to ship builds to AWS
   - Disable 
   - Ensure all historical cache data is now available on S3

## Later

- Publish cache data back to GitHub?
  - We could build a simple scheduled event to pull down the data from S3 and sync it to GitHub, ensuring an off-network backup as well as a developer-friendly way off accessing our cache dataset

**Comment by lazd** [comment]
I think this moves us in the right direction without interrupting the current developer workflow. The work done in #224 is non-invasive and will give us a chance to run the existing deployment side-by-side with AWS until we're ready to switch.

As we do the work in step 4, we'll get to re-approach what we do with errors, warning, scraper failures, and cross-check reports. This will be a welcome chance for us increase quality, facilitate verification & validation, and build better data through improvements to how we handle lots.

Finally, as we implement step 5, the local developer workflow will actually become simpler than it is right now, so we shouldn't be complicating life for potential and existing contributors.

+💯!!!11!11!1111!!1one

**Comment by ryanblock** [comment]
Followup: https://github.com/lazd/coronadatascraper/issues/295

**Comment by ryanblock** [comment]
Closing in favor of https://github.com/covidatlas/coronadatascraper/issues/782
